File: app/assets/javascripts/admin/caller.js
$(document).ready(function() {
    const checkbox_labels = {
        program: "J’ai explicitement informé la famille que ne plus vouloir des livres, des appels ou des SMS revenait à mettre fin à l’accompagnement et elle m’a confirmé vouloir arrêter OU La famille m’a explicitement demandé à ne plus recevoir l’accompagnement 1001mots.",
        popi: "J’ai eu des éléments factuels qui prouvent que la famille a un niveau socio-économique élevé (bac+5, revenu très élevé...)",
        professional: "J’ai eu des éléments factuels qui prouvent que la famille est en fait un.e professionnel.le de santé qui souhaite tester l’accompagnement dans le cadre de son travail",
        problematic_case: "J’ai validé l’arrêt de cette famille avec une coordinatrice"
    }
    let radios = document.querySelectorAll("input[type='radio']");
    let checkbox = document.querySelector("input[type='checkbox']");
    let checkbox_label = document.getElementById('checkbox-label');
    let form_details = document.getElementById('form-details');
    let validation = document.querySelector("input[type='submit']");
    form_details.style.display = 'none';
    validation.disabled = true;
    radios.forEach(function(radio) {
        radio.addEventListener('change', function() {
            if(radio.checked) {
                form_details.style.display = 'block';
                checkbox_label.textContent = checkbox_labels[radio.value]
            } else {
                form_details.style.display = 'none';
            }
        })
    });
    checkbox.addEventListener('change', function() {
        if (checkbox.checked) {
            validation.disabled = false;
        } else {
            validation.disabled = true;
        }
    });
});

File: app/admin/caller.rb
ActiveAdmin.register_page "Caller" do
  menu false

  content do

    form action: admin_caller_stop_support_path, method: :post, id: "stop-support-form" do |f|
      f.input :authenticity_token, type: :hidden, name: :authenticity_token, value: form_authenticity_token
      f.input :supporter_id, type: :hidden, name: :supporter_id, value: params[:supporter_id]
      f.input :child_support_id, type: :hidden, name: :child_support_id, value: params[:child_support_id]

      div do
        label class: 'label-for-group' do
          "Raison de l'arrrêt de l'accompagnement"
        end
        hr
        div class: 'radio-input' do
          f.input :stop_reason, type: :radio, value: :program, name: :reason
          label "La famille que j’accompagne ne veut pas du programme complet"
        end
        div class: 'radio-input' do
          f.input :stop_reason, type: :radio, value: :popi, name: :reason
          label "La famille que j’accompagne a un niveau socio-économique élevé (BAC+5)"
        end
        div class: 'radio-input' do
          f.input :stop_reason, type: :radio, value: :professional, name: :reason
          label "La famille que j’accompagne est en fait un.e professionnel.le de santé qui souhaite tester l’accompagnement"
        end
        div class: 'radio-input' do
          f.input :stop_reason, type: :radio, value: :problematic_case, name: :reason
          label "La famille que j’accompagne me pose problème : je demande l’arrêt de l’accompagnement sur avis de ma coordinatrice"
        end
      end
      div id: 'form-details' do
        hr
        div id: 'checkbox-input' do
          f.input :validation, type: :checkbox
          label id: 'checkbox-label' do
            ''
          end
        end
        hr
        div do
          label class: 'label-for-group' do
            'Détaillez les éléments factuels en quelques mots'
          end
          textarea name: :details
        end
      end
      div class: "actions" do
        div class: "action input_action" do
          input type: "submit", value: "Valider", data: { confirm: 'Je confirme vouloir arrêter l’accompagnement pour cette famille' }
        end
      end
    end
  end

  page_action :stop_support, method: :post do
    stop_support_service = ChildSupport::CallerStopSupportService.new(
      params[:supporter_id],
      params[:child_support_id],
      params[:reason],
      params[:details]
    ).call

    if stop_support_service.error.nil?
      redirect_to admin_child_support_path(params[:child_support_id]), notice: 'Accompagnement arrêté et message envoyé'
    else
      redirect_to admin_child_support_path(params[:child_support_id]), alert: stop_support_service.error
    end
  end
end
